network.c

```diff
@@ -1402,6 +1402,29 @@ struct iio_context * network_create_context(const char *host)
 #endif
 	{
 		ret = getaddrinfo(host, IIOD_PORT_STR, &hints, &res);
+		/*
+		 * It might be an avahi hostname which means that getaddrinfo() will only work if
+		 * nss-mdns is installed on the host and /etc/nsswitch.conf is correctly configured
+		 * which might be not the case for some minimalist distros. In this case,
+		 * as a last resort, let's try to resolve the host with avahi...
+		 */
+		if (ret) {
```

Comment:
With the latest master, you can write: `if (HAVE_DNS_SD && ret)`. Then you can remove the static inline  `dnssd_resolve_host` fallback in network.h.

Reply:
Yeah I know :). If we are going to take this, I definitely have to re-apply all of this on top of the new master. But I'm only going to have that work if this is to be merged :smile:

Reply:
Well, I don't see why I wouldn't merge it, if it's done properly and has been tested. Go for it!

Reply:
As it stands, it does not work...

```
[nsa@nsa libiio]$ ./net
ERROR: Unable to create Avahi DNS-SD relsover: Invalid host name
ERROR: Unable to create Avahi DNS-SD relsover: Invalid host name
```

However, I can tell that `avahi-resolve` has an address option that just uses another [implementation](https://github.com/rockchip-android/external-avahi/blob/e67ae65eb3c48969c16751feccfa9daf74b0f406/avahi-utils/avahi-resolve.c#L306). So, in theory, we could support it but we would need more complexity [basically just checking if we have an IP]. For the localhost, I'm getting the same error and I'm not sure if there's a way for it. So, IMO I would just stick with the fallback otherwise we might be breaking some corner case that we are not seeing...

Reply:
Maybe we would not even need to do the IP check... but the localhost is still a question mark

Reply:
Ups, just realized now that this is the wrong thread :facepalm:

Reply:
Alright, green light for me then.